addons/hr_timesheet/tests/test_timesheet.py

```diff
@@ -810,3 +810,18 @@ def test_analytic_plan_timesheet_change_analytic_account(self):
         timesheet.task_id = self.task2
         self.assertEqual(analytic_account, timesheet.account_id)
         self.assertEqual(analytic_account, timesheet[f'{analytic_plan._column_name()}'])
+
+    def test_log_timesheet_with_user_has_two_employees_from_different_companies(self):
+        company_2 = self.env['res.company'].create({'name': 'Company 2'})
+        self.env['hr.employee'].with_company(company_2).create({
+            'name': 'Employee 2',
+            'user_id': self.user_manager.id,
+        })
+        project = self.env['project.project'].create({
+            'name': 'Test project',
+        })
+        Timesheet = self.env['account.analytic.line']
+        Timesheet.create({
+            'project_id': project.id,
```

Comment:
```suggestion
        Timesheet = self.env['account.analytic.line'].create({
            'project_id': self.project.id,
```
Also add an assert to check the record values of the timesheet